src/lib/comms/sol-oic-server.c

```diff
@@ -754,6 +681,21 @@ sol_oic_server_add_resource(const struct sol_oic_resource_type *rt,
     return NULL;
 }
 
+SOL_API struct sol_oic_server_resource *
+sol_oic_server_add_resource(const struct sol_oic_resource_type *rt,
+    const void *handler_data, enum sol_oic_resource_flag flags)
```

Comment:
I would name this function to: sol_oic_server_resource_auto and the sol_oic_server_resource_full to sol_oic_server_resource.